bravado/mapping/marshal.py

```diff
@@ -0,0 +1,63 @@
+import urllib
+
+import jsonschema
+
+from bravado.exception import SwaggerError
+
+# TODO: generalize for any Swagger Spec object, not just parameters
+
+def validate_primitive(param, value):
+    if value is None and param.has_default():
+        value = param.default
+
+    if param.required and value is None:
+        raise SwaggerError('Parameter {0} cannot be null'.format(param.name))
+
+    jsonschema.validate(value, param.jsonschema)
+
+    # TODO: transform
+    return value
+
+
+def marshal_primitive(param, value, request):
+    if param.location == 'path':
+        token = u'{%s}' % param.name
+        request['url'] = request['url'].replace(token, urllib.quote(unicode(value)))
```

Comment:
I really want to say that this can be done with `string.format()`, but in order to make that work, this whole thing would have to change. Instead of doing param at a time, it would have to group everything by `param.location`, then validate every param separately, and finally build the request field all together at once with all param/values pairs for the same `param.location`.

This is something I wanted to do with the original swagger-py, but I suspect you aren't going to want to do that at this stage, so I guess we can leave it for now.

Another plus I see with this approach is that it makes this code immutable. We don't have to pass a request dict around and mutate it all over.